openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -222,7 +237,7 @@ def get_cohort(user, course_key, assign=True):
 
     user.course_groups.add(cohort)
 
-    return cohort
+    return request_cache.data.setdefault(cache_key, cohort)
```

Comment:
I'm actually not sure how the request cache works in the context of management commands. If it's just a matter of not getting cleared out until the end, we'd need to take a look at what's being cached. A simple mapping of users to groups probably won't hurt us much, even for large courses -- we'd just have to make sure there aren't key conflicts between users, which is a good practice anyway.